### PR TITLE
Add architectures and packages selectors

### PR DESCRIPTION
**Pull request by dheyay**
## Why is this needed?

We needed APT News selectors to only display messages on systems with certain package versions or certain architectures. This adds selectors for both architectures and packages to filter APT news.

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

File: uaclient/apt_news.py
Comment (orndorffgrant) [inline uaclient/apt_news.py:115]: We should check that there are three items in this list before doing this. We can log something and return False if there are not three items.

File: uaclient/apt_news.py
Comment (orndorffgrant) [inline uaclient/apt_news.py:139]: Style suggestion
```suggestion
                if any(
                    [
                        (version_comparison == 0 and version_operator in ["==", "<=", ">="]),
                        (version_comparison < 0 and version_operator in ["<", "<="]),
                        (version_comparison > 0 and version_operator in [">", ">="]),
                    ]
                ):
```

File: uaclient/apt_news.py
Comment (orndorffgrant) [inline uaclient/apt_news.py:116]: Optional suggestion: invert this if statement to return early
```suggestion
            if not apt.is_installed(package_name):
                LOG.info("%s not installed", package_name)
                return False
```

**Review by orndorffgrant — COMMENTED** [review]
Looking good!

Please add cases to the test in apt_messages.feature to cover this new functionality.

File: uaclient/apt_news.py
Comment (dheyay) [reply uaclient/apt_news.py:139]: Looks much better than the original change haha, thanks @orndorffgrant

**Comment by renanrodrigo** [comment]
This is not building because there is an extra comma in `apt_news.py:54`

File: features/apt_messages.feature
Comment (lucasmoura) [inline features/apt_messages.feature:1012]: I think we should test this on all LTS releases as well

File: uaclient/apt_news.py
Comment (lucasmoura) [inline uaclient/apt_news.py:128]: instead of looking at the packages list, we can instead use the `get_pkg_version` function directly

File: uaclient/apt_news.py
Comment (lucasmoura) [inline uaclient/apt_news.py:149]: I think we can update the logic to a `not any` and just return `False` here.
Then we don't need the `is_package_matched` variable anymore

File: uaclient/apt_news.py
Comment (lucasmoura) [reply uaclient/apt_news.py:128]: If we do that, we don't even need the `installed_packages` anymore

Comment (dheyay) [reply uaclient/apt_news.py]: Thanks @lucasmoura , seems like a good idea.
Updated!

File: uaclient/apt_news.py
Comment (lucasmoura) [inline uaclient/apt_news.py:121]: I think we need to think of the situation where this function returns `None` (when the package doesn't exist yet)

Comment (dheyay) [reply uaclient/apt_news.py:121]: Would that not be flagged be the installed package check and return False earlier?

File: features/apt_messages.feature
Comment (dheyay) [reply features/apt_messages.feature]: Updated the test to support all LTS releases @lucasmoura

File: uaclient/apt_news.py
Comment (lucasmoura) [reply uaclient/apt_news.py:121]: Yes, you are totally right. But now, looking at the code, I think it might be better if we don't have that call to `is_installed` at all.

My rationale here is that this function will fetch all installed packages and then perform a search on that list.
For the `get_pkg_version` we directly check the package in the apt cache, which should be faster.

We would then return false only if `get_pkg_version` returns `None`.

What do you think ?

Comment (dheyay) [reply uaclient/apt_news.py:121]: Oh yes, actually, that makes a lot of sense. That would be a good change, thanks @lucasmoura

File: features/apt_messages.feature
Comment (orndorffgrant) [inline features/apt_messages.feature:885]: Many of these `stdout matches regexp` steps can be replaced by the stricter `stdout contains substring:` step

**Review by orndorffgrant — APPROVED** [review]
This looks great! just a nit on the tests.

And the spec isn't actually approved yet so we shouldn't merge this for now.

**Comment by dheyay** [comment]
Updated @orndorffgrant

**Comment by orndorffgrant** [comment]
The CI failure in the apt news test was strange and I couldn't reproduce it locally. If it happens again, we can add a step that clears the pro apt cache which should prevent that message from occurring during the test.

**Review by lucasmoura — APPROVED** [review]

**Comment by renanrodrigo** [comment]
Well I had understood that this would wait for a proper spec approval
But we can iterate in further PRs if things change there I guess (:

**Pull request merged**
